Modules/Scripted/DICOMPlugins/DICOMScalarVolumePlugin.py

```diff
@@ -224,10 +237,34 @@ def examineFiles(self, files):
             for tag in subseriesTags:
                 value = slicer.dicomDatabase.fileValue(file, self.tags[tag])
                 value = value.replace(",", "_")  # remove commas so it can be used as an index
+
                 if tag not in subseriesValues:
                     subseriesValues[tag] = []
-                if not subseriesValues[tag].__contains__(value):
+
+                if tag in vectorTags:
+                    if value != "":
+                        vector = numpy.array([float(element) for element in value.split("\\")])
+
+                        found = False
+                        for subseriesValue in subseriesValues[tag]:
+                            subseriesVector = numpy.array([float(element) for element in subseriesValue.split("\\")])
+
+                            # The 'eps' value default is set to 1.e-6, which reproduces the same behavior as
+                            # the ITK numeric comparison logic (i.e. absolute difference).
+                            # Reference:
+                            #   Class: ITK/Modules/Numerics/Optimizersv4/include/itkObjectToObjectMetric.hxx
+                            #   Method: VerifyDisplacementFieldSizeAndPhysicalSpace
+                            #   URL: https://github.com/hinerm/ITK/blob/master/Modules/Numerics/Optimizersv4/include/itkObjectToObjectMetric.hxx#L471-L475.
+                            if numpy.allclose(vector, subseriesVector, rtol = 0., atol = eps):
+                                found = True
+                                value = subseriesValue
+                                break
+
+                        if not found:
+                            subseriesValues[tag].append(value)
+                elif not subseriesValues[tag].__contains__(value):
                     subseriesValues[tag].append(value)
```

Comment:
To streamline the overall logic and reduce complexity, I suggest to revisit the approach by:
* adding a `TagValueList` custom type that would be a subclass of `list` with its `__contains__` function overridden to support comparing `np.ndarray`
* adding a function `tagValueToVector`

using something like this:

```python
        subseriesFiles = {}
        subseriesValues = {}
        for file in allFilesLoadable.files:
            # check for subseries values
            for tag in subseriesTags:
                if tag not in subseriesValues:
                    subseriesValues[tag] = TagValueList(epsilon=eps)

                value = slicer.dicomDatabase.fileValue(file, self.tags[tag])

                if tag in subseriesTagsToVectorValues:
                    value = tagValueToVector(value)
                else:
                    value = value.replace(",", "_")  # remove commas so it can be used as an index

                if not subseriesValues[tag].__contains__(value):
                    subseriesValues[tag].append(value)

                if (tag, value) not in subseriesFiles:
                    subseriesFiles[tag, value] = []
                subseriesFiles[tag, value].append(file)
```


with


```
def tagValueToVector(value):
    return numpy.array([float(element) for element in value.split("\\")])
```

and 

```python
class TagValueList(list):
    def __init__(self, epsilon):
        self.epsilon = epsilon
    def __contains__(self, value):
        if isinstance(value, numpy.ndarray):
            return numpy.allclose(vector, subseriesVector, rtol=0.0, atol=self.epsilon):
        return super().__contains__(value):
```

Reply:
ok thanks for the feedback. I agree and I have implemented it in https://github.com/Slicer/Slicer/pull/7362/commits/3b7f8cbe25e39a50dc74c6e17529dcf1e56514ce
however I had to modify a little bit the design. Please have a look.

Reply:
I'm not following - why are we using `__contains__` and not `in`?

Reply:
This should indeed be simplified and could be done in this PR or a follow-up one.

The use of `__contains__` was originally introduced in https://github.com/Slicer/Slicer/commit/f0b2690c328fc0cfc248907d35ab845c3e7c4f4d and was likely on oversight.

Reply:
> was likely on oversight.

Yes, that was me in the old days - good time to fix it now : )